### DISCUSSION TITLE
Resolving Domain Ownership Conflicts in Vercel Multi-Tenancy

### DISCUSSION THREAD

**Discussion by safdar-azeem** [vercel/community#4027] [Help]
### Summary

I'm encountering an issue related to domain verification while working on a multi-tenancy web project. 

Here's the scenario: 

When a user attempts to add a custom domain to their account, but they don't actually own the domain or haven't verified it through DNS records, things get tricky. Later, another user who genuinely owns the domain tries to set it up and configure the necessary DNS records in his provider.

In this scenario, it becomes crucial to determine which user has successfully configured the domain so that it can be associated with their account in the database.

It's also happening in the Vercel-hosted multi-tenancy template project, https://app.vercel.pub/

I'm in need of guidance on how to tackle this problem.  Any insights or suggestions would be greatly appreciated.

### Steps to Reproduce

.

**Comment by amyegan** [comment]
Hey @safdar-azeem. When you [add a domain to a project](https://vercel.com/docs/rest-api/endpoints#add-a-domain-to-a-project) using `POST /v10/projects/{idOrName}/domains`, you'll see a `verified` true/false status with the response and a list of `verification` challenges if one is needed.

You can also [verify project domain](https://vercel.com/docs/rest-api/endpoints#verify-project-domain) with `POST /v9/projects/{idOrName}/domains/{domain}/verify` and [check a domain's configuration](https://vercel.com/docs/rest-api/endpoints#get-a-domain-s-configuration) with `GET /v6/domains/{domain}/config`

**Reply by safdar-azeem** [reply]
Yeah i am using the `POST /v10/projects/{idOrName}/domains` endpoint to add domains to my project, and I'm getting a 'verified' status in the response, but I'm not receiving verification challenges. Here's what I'm getting:

{
    "name": "",
    "apexName": "",
    "projectId": "",
    "redirect": null,
    "redirectStatusCode": null,
    "gitBranch": null,
    "updatedAt": "",
    "createdAt": "",
    "verified": true
}

Is there something I'm missing or should be doing differently to get the verification challenges?

**Reply by amyegan** [reply]
`verified: true` indicates that a verification challenge isn't needed
